components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/DefaultIDTokenBuilder.java

```diff
@@ -792,5 +798,44 @@ private boolean isValidIdToken(JWTClaimsSet jwtClaimsSet) {
     private long getIDTokenExpiryInMillis() {
         return OAuthServerConfiguration.getInstance().getOpenIDConnectIDTokenExpiryTimeInSeconds() * 1000L;
     }
-}
 
+    public void setAddtionalClaims(OAuthTokenReqMessageContext tokenReqMsgCtxt,
+                                           OAuth2AccessTokenRespDTO tokenRespDTO, JWTClaimsSet jwtClaimsSet)
+            throws IdentityOAuth2Exception {
+        List<ClaimProvider> claimProviders = OpenIDConnectServiceComponentHolder.getClaimProviders();
+        if (!claimProviders.isEmpty()) {
+            for (ClaimProvider claimProvider : claimProviders) {
```

Comment:
This for loop and the one in the other setAddtionalClaims seem to have lot of common code. See whether you can refactor them. Also, try to use Java8 lamba here if possible